backend/app/assets/javascripts/spree/backend/admin.js

```diff
@@ -53,24 +53,48 @@ $.fn.radioControlsVisibilityOfElement = function(dependentElementSelector){
 }
 
 var handle_date_picker_fields = function(){
-  $('.datepicker').datepicker({
-    dateFormat: Spree.t('date_picker.js_format', { default: 'yy/mm/dd' }),
-    dayNames: Spree.t('abbr_day_names'),
-    dayNamesMin: Spree.t('abbr_day_names'),
-    firstDay: parseInt(Spree.t('date_picker.first_day', { default: '0' }), 10),
-    monthNames: Spree.t('month_names'),
-    prevText: Spree.t('previous'),
-    nextText: Spree.t('next'),
-    showOn: "focus"
+  $('.datepicker').flatpickr({
+    allowInput: true
   });
 
-  // Correctly display range dates
-  $('.date-range-filter .datepicker-from').datepicker('option', 'onSelect', function(selectedDate) {
-    $(".date-range-filter .datepicker-to" ).datepicker( "option", "minDate", selectedDate );
-  });
-  $('.date-range-filter .datepicker-to').datepicker('option', 'onSelect', function(selectedDate) {
-    $(".date-range-filter .datepicker-from" ).datepicker( "option", "maxDate", selectedDate );
-  });
+  // Handle range dates
+  if ($('.date-range-filter .datepicker-from, .date-range-filter .datepicker-to').length) {
```

Comment:
What about trying to create a plugin for this behavior?

Reply:
I actually already implemented the former behavior, and I saved that code in the case it was decided to preserve that behavior. If requested I can switch to that.

Reply:
I meant moving this behavior you implemented into a new `flatpickr` plugin that we'll include instead of putting this code here directly but it's just an idea for better isolate this functionality that we could choose to replace later. Anyway maybe it's better to wait for other core team members opinion on flatpickr before putting too much effort here. Thanks!

Reply:
Yes, we could talk together about that in the issue discussion maybe? I don't know if all the points I made above in the pull request comment will be considered :flushed: